ConcatenateMatrices.java

```diff
@@ -3,33 +3,69 @@
 
 public class ConcatenateMatrices {
 
-    public static void main(String[] args) throws IOException {
-        Scanner scanner = new Scanner(new File("vstup.txt"));
-        PrintStream output = new PrintStream("vystup.txt");
+    public static void main(String[] args){
 
-        int m = scanner.nextInt();
-        int n = scanner.nextInt();
+        Scanner scanner = null;
+        PrintStream output = null;
+        try {
+            scanner = new Scanner(new File("vstup.txt"));
+            output = new PrintStream("vystup.txt");
+        }
+        catch (Exception e){
+            System.out.println("Problem s nacitanim/vytvorenim vstupneho/vystupneho suboru.");
+            if(scanner != null) scanner.close();
+            if(output != null) output.close();
+            return;
+        }
+
+        int m, n;
```

Comment:
In general, using non-initialized variables (they are initialized to zero, but that is a meaningless value) has drawbacks. If someone later adds code between lines 21 and 22, it might introduce a bug. (Such a scenario is too far-fatched in this trivial example, but not every compiler is good at reporting such potentially dangerous use, consider e.g. javascript.)

This problem would entirely disappear if the code was split into three separate parts:
(1) reading and parsing input;
(2) validating input (e.g. positive values for m, n);
(3) subsequent computations on valid input.